crates/optimism/rpc/src/eth/block.rs

```diff
@@ -60,13 +60,18 @@ where
                         timestamp,
                     };
 
+                    // We must clear this cache as different L2 transactions can have different
+                    // L1 costs. A potential improvement here is to only clear the cache if the
+                    // new transaction input has changed, since otherwise the L1 cost wouldn't.
+                    l1_block_info.clear_tx_l1_cost();
```

Comment:
-.-
I knew that introducing this behaviour would end causing problems somewhere